tools/dotnet-linker/Steps/GenerateReferencesStep.cs

```diff
@@ -0,0 +1,69 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+using Mono.Cecil;
+
+using Xamarin.Bundler;
+using Xamarin.Linker;
+
+namespace Xamarin {
+
+	public class GenerateReferencesStep : ConfigurationAwareStep {
+		private Symbols required_symbols = new Symbols ();
+		
+		protected override string Name { get; } = "Generate References";
+		protected override int ErrorCode { get; } = 2320;
+
+		protected override void TryEndProcess ()
+		{
+			base.TryEndProcess ();
+
+			var items = new List<MSBuildItem> ();
+			var file = Path.Combine (Configuration.CacheDirectory, $"references.mm");
+			if (Configuration.Target.GenerateReferencingSource (file, required_symbols) != null) {
+				var item = new MSBuildItem { Include = file };
+				items.Add (item);
+			}
+			Configuration.WriteOutputForMSBuild ("_ReferencesFile", items);
+		}
+
+		protected override void TryProcessAssembly (AssemblyDefinition assembly)
```

Comment:
I think we should re-use the existing ListExportedSymbols step instead of duplicating the logic here, because sooner or later we'll also need the rest of the logic in that step.

Reply:
That's what I initially planned to do. Then I discovered that not all of the logic is applicable anymore in the .NET 6 context so I went with an approach that kept the code to absolute minimum. I am happy to rewrite it to use `ListExportedSymbols` but I need to study the `PInvokeWrapperGenerator` and related code first. Also, all the code for `System.*.Native.*` libraries is not necessary in the macOS context since the libraries are linked dynamically.